snakemake/satija/src/convert_h5ad_to_zarr.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+
+from anndata import read_h5ad
+import pandas as pd
+import scipy.cluster
+import zarr
+from numcodecs import Zlib
+
+
+def h5ad_to_zarr(input_file, output_file):
+    gexp = read_h5ad(input_file)
+    gexp_arr = gexp.X
+    gexp_df = gexp.to_df()
+
+    # Re-scale the gene expression values between 0 and 255 (one byte ints).
+    gexp_arr_min = gexp_arr.min()
+    gexp_arr_max = gexp_arr.max()
+    gexp_arr_range = gexp_arr_max - gexp_arr_min
+    gexp_arr_ratio = 255 / gexp_arr_range
+    gexp_norm_arr = (gexp_arr - gexp_arr_min) * gexp_arr_ratio
```

Comment:
Is it still floats at this point?

Reply:
I noticed this too when looking back at the other PR - I saw some sliders on the Vitessce PR so how does that work?

Reply:
Yes the casting does not happen until the `zarr.open` line where I pass `dtype='uint8'`

Reply:
I did add a color scale slider, it works fine with the integer data

Reply:
I am wondering then why not take it further and use uint16 or a 32 bit float (i.e scaling to 0-1) if we have a slider.  Is this standard in other heat maps?

Reply:
Wouldn't that double or quadruple the file sizes though? These zarr directories are ~15 MB and I am requesting all chunks at the same time in the browser

Reply:
I just tried increasing to uint16 for one of the datasets and file size increased from 15.4 MB to 22.9 MB

Reply:
As far as other heatmaps:
- clustergrammer: TSV or in-memory pandas df via python https://clustergrammer.readthedocs.io/matrix_format_io.html
    - the largest file I see in their example files is a 3.1 MB TSV with shape (3406, 458) https://github.com/MaayanLab/clustergrammer/blob/master/txt/KEA_matrix.txt

- plotly
    - there is an example of a heatmap that takes a JPG image as input https://plotly.com/javascript/heatmap-webgl/#webgl-heatmaps
    - there are other examples here https://plotly.com/javascript/heatmaps/

- sanddance: loads a 3.59 MB TSV https://sanddance.js.org/app/

I don't see any interactive color scales

Reply:
This is a demo with the color scale slider https://s3.amazonaws.com/vitessce-data/demos/2020-07-21/0ccb14d/index.html?dataset=linnarsson-2018&theme=dark